polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -446,7 +443,16 @@ fn handle_parent_process(
 				},
 			}
 		},
-		_ => Err(PrepareError::Panic("child finished with unknown status".to_string())),
+		Ok(WaitStatus::Exited(_, libc::EXIT_FAILURE)) => {
+			Err(PrepareError::Panic("child exited with failure".to_string()))
+		},
+		Ok(WaitStatus::Exited(_, exit_status)) => {
+			Err(PrepareError::Panic(format!("child exited with unexpected status {}", exit_status)))
+		},
+		Ok(_) => {
+			Err(PrepareError::Panic("child ended unexpectedly".to_string()))
+		}
+		Err(err) => Err(PrepareError::Panic(err.to_string()))
```

Comment:
Hmm, an attacker can make the child process return any exit status it wants. So we should treat all of these cases the same way (like we were before when we matched on `_`). We should reserve `JobDied` for if the child process actually dies, and this isn't a panic, so a different error variant is warranted here. Maybe just simply `IoErr`.